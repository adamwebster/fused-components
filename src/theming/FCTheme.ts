import React from 'react';

export interface Theme {
    theme?: string,
    children?: any,
}

export const FCTheme = React.createContext<Theme | null>(null);

export const FCThemeProvider = FCTheme.Provider;

export const FCThemeConsumer = FCTheme.Consumer;