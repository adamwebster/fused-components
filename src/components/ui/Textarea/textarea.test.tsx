import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';
import { Textarea } from './index';
import userEvent from '@testing-library/user-event';
import { FCThemeProvider } from '../../../theming/FCTheme';

afterEach(cleanup);

describe('Textarea Input', () => {
  test('Renders the Textarea component', () => {
    const { getByPlaceholderText } = render(<Textarea placeholder="Test placeholder" />);
    expect(getByPlaceholderText('Test placeholder')).toBeInTheDocument();
  });
  test('Changes the value on change', () => {
    let initialValue = '';
    const onChange = jest.fn(e => {
      initialValue = e.target.value;
    });

    const { getByPlaceholderText, rerender } = render(
      <Textarea value={initialValue} onChange={e => onChange(e)} placeholder="Test placeholder" />,
    );

    const input = getByPlaceholderText('Test placeholder');
    userEvent.type(input, 'Hello World');
    expect(onChange).toBeCalledTimes(11);
    rerender(<Textarea value={initialValue} onChange={e => onChange(e)} placeholder="Test placeholder" />);
    expect(input).toHaveValue('Hello World');
    expect(initialValue).toBe('Hello World');
  });

  test('Renders the Input as disabled', () => {
    const { getByPlaceholderText } = render(<Textarea disabled placeholder="Test placeholder" />);
    expect(getByPlaceholderText('Test placeholder')).toBeDisabled();
  });

  test('Is readonly', () => {
    const { getByPlaceholderText } = render(<Textarea readOnly placeholder="Test placeholder" />);
    expect(getByPlaceholderText('Test placeholder').getAttribute('readonly')).toBe('');
  });
  test('Sets the id', () => {
    const { getByPlaceholderText } = render(<Textarea id="Input control" placeholder="Test placeholder" />);
    expect(getByPlaceholderText('Test placeholder').getAttribute('id')).toBe('Input control');
  });

  test('Autocomplete can be set to off', () => {
    const { getByPlaceholderText } = render(
      <Textarea id="Input control" autoComplete="off" placeholder="Test placeholder" />,
    );
    expect(getByPlaceholderText('Test placeholder').getAttribute('autocomplete')).toBe('off');
  });

  test('Has the correct styles when in dark mode', () => {
    const { container } = render(
      <FCThemeProvider value={{ theme: 'dark' }}>
        <Textarea placeholder="Test placeholder" />
      </FCThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
