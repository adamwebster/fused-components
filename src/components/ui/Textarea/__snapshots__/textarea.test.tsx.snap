// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea Input Has the correct styles when in dark mode 1`] = `
.c0 {
  border-radius: 5px;
  color: inherit;
  padding: 10px;
  width: 100%;
  box-sizing: border-box;
  -webkit-appearance: none;
  resize: vertical;
  outline: none;
  border: solid 1px #627d98;
  background-color: #243b53;
}

.c0:focus {
  border-color: #627d98;
  box-shadow: 0 0 6px #627d98;
}

<div>
  <textarea
    class="c0"
    placeholder="Test placeholder"
  />
</div>
`;
